lib/ledger_sync/adaptors/netsuite/record/metadata.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module LedgerSync
+  module Adaptors
+    module NetSuite
+      module Record
+        class Metadata
+          BASE_PATH = 'metadata-catalog/record'
+
+          attr_reader :adaptor,
+                      :record
+
+          def initialize(adaptor:, record:)
+            @adaptor = adaptor
+            @record = record
+          end
+
+          def http_methods
+            @http_methods ||= begin
+              ret = []
+
+              http_methods_response.body['paths'].each do |path, path_data|
```

Comment:
I get that we're doing 2 dimensional array here, but isnt doing `.map` and `.map` and then `.flatten` still nicer?